projects/plugins/boost/app/modules/cache/Boost_Cache.php

```diff
@@ -254,4 +272,208 @@ public function is_backend() {
+	 */
+	public function delete_on_comment_edit( $comment_id, $commentdata ) {
+		$post = get_post( $commentdata['comment_post_ID'] );
+
+		if ( (int) $commentdata['comment_approved'] === 1 ) {
+			$this->delete_cache_for_post( $post );
+		}
+	}
+
+	/*
+	 * After a comment is posted, delete the cache for the post if the comment is approved.
+	 * If the comment is not approved, only delete the cache for this post for this visitor.
+	 *
+	 * @param int $comment_id - The id of the comment.
+	 * @param int $comment_approved - The approval status of the comment.
+	 * @param array $commentdata - The comment data.
+	 */
+	public function delete_on_comment_post( $comment_id, $comment_approved, $commentdata ) {
+		$post = get_post( $commentdata['comment_post_ID'] );
+
+		/*
+		 * If a comment is not approved, we only need to delete the cache for
+		 * this post for this visitor so the unmoderated comment is shown to them.
+		 */
+		if ( $comment_approved !== 1 ) {
+			$this->storage->invalidate_single_visitor( $this->normalize_request_uri( get_permalink( $post->ID ) ), $this->request_parameters );
+			return;
+		}
+
+		$this->delete_cache_for_post( $post );
+	}
+
+	/*
+	 * Returns true if the post is published or private.
+	 *
+	 * @param string $status - The status of the post.
+	 * @return bool
+	 */
+	private function is_published( $status ) {
+		return $status === 'publish' || $status === 'private';
+	}
+
+	/*
+	 * Delete the cached post if it transitioned from one state to another.
+	 *
+	 * @param string $new_status - The new status of the post.
+	 * @param string $old_status - The old status of the post.
+	 * @param WP_Post $post - The post that transitioned.
+	 */
+	public function delete_on_post_transition( $new_status, $old_status, $post ) {
+		if ( ! $this->is_visible_post_type( $post ) ) {
+			return;
+		}
+
+		error_log( "delete_on_post_transition: $new_status, $old_status, {$post->ID}" ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+
+		// Don't delete the cache for posts that weren't published and aren't published now
+		if ( ! $this->is_published( $new_status ) && ! $this->is_published( $old_status ) ) {
+			error_log( 'delete_on_post_transition: not published' ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+			return;
+		}
+
+		error_log( "delete_on_post_transition: deleting post {$post->ID}" ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+
+		$this->delete_cache_for_post( $post );
+		$this->delete_cache_for_post_terms( $post );
+		$this->delete_front_page_cache( $post );
+	}
+
+	/*
+	 * Deletes cache files for the given post.
+	 *
+	 * @param WP_Post $post - The post to delete the cache file for.
+	 * @param bool $all - If false, only delete the cache file for the post, not the front page cache.
+	 */
+	public function delete_cache_for_post( $post ) {
+		static $already_deleted = -1;
+		if ( $already_deleted === $post->ID ) {
+			return;
+		}
```

Comment:
If we can end up invalidating the same post multiple times in a single PHP execution, could we also end up deleting multiple posts, multiple times: Is it worth storing an array of post_ids that have been invalidated this PHP execution to prevent re-deleting?

I don't know the answer to this question, btw - so if the answer is "no" that's ok.

Reply:
I copied this from WPSC where there are more actions fired for deleting cache files, but since there are a few comment actions hooked, it's still needed.
I think it's ok to have just one. When I logged the deletes I didn't see the same post being deleted twice with this in place.